src/main/java/space/obminyashka/items_exchange/service/impl/LocationServiceImpl.java

```diff
@@ -130,24 +130,26 @@ public String createParsedLocationsFile(List<RawLocation> creatingData) throws I
 
     @Override
     public String createParsedLocsFile(List<LocationRaw> creatingData) throws IOException {
+        creatingData.sort(Comparator.comparing(LocationRaw::getAreaEn));
+
         Set<Area> areas = new HashSet<>();
         Set<District> districts = new HashSet<>();
         Set<City> cities = new HashSet<>();
 
         for (LocationRaw location : creatingData) {
-            if (!location.getAreaUa().isBlank() && !location.getAreaEn().isBlank()) {
-                Area area = new Area(location);
-                areas.add(area);
-
-                if (!location.getDistrictUa().isBlank() && !location.getDistrictEn().isBlank()) {
-                    District district = new District(location, area);
-                    districts.add(district);
-
-                    if (!location.getCityUa().isBlank() && !location.getCityEn().isBlank()) {
-                        cities.add(new City(location, district));
-                    }
-                }
-            }
+            Area area = new Area(location);
+            List<LocationRaw> locationsWithDistrictInSelectedArea = creatingData.stream()
+                    .sorted(Comparator.comparing(LocationRaw::getDistrictEn))
+                    .filter(locationRaw -> locationRaw.getAreaEn().equals(area.getNameEn()))
```

Comment:
In my previous commit, I added equals and hash code for Area, District, and City that would allow you to distinguish similar objects.
Why are you doing the same things but comparing only the field?